core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -263,6 +263,9 @@ static void init(String[] args) throws Throwable {
         INSTANCE = new Bootstrap();
 
         boolean foreground = !"false".equals(System.getProperty("es.foreground", System.getProperty("es-foreground")));
+        // Clear these properties because we are done with them. No need to let them leak into the Settings.
+        System.clearProperty("es.foreground");
```

Comment:
If it is just trying to avoid adding a "foreground" setting, then I think the solution is to not use `es.` prefix? Then it won't be put into settings.